pkg/controller/useraccountstatus/useraccount_status_controller.go

```diff
@@ -88,33 +90,41 @@ func (r *ReconcileUserAccountStatus) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	err, mur := r.updateMasterUserRecord(userAcc)
+	mur, err := r.updateMasterUserRecord(logger, userAcc)
 	if err != nil {
-		reqLogger.Error(err, "unable to update the master user record", "MasterUserRecord", mur, "UserAccount", userAcc)
+		logger.Error(err, "unable to update the master user record", "MasterUserRecord", mur, "UserAccount", userAcc)
 		return reconcile.Result{}, err
 	}
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileUserAccountStatus) updateMasterUserRecord(userAcc *toolchainv1alpha1.UserAccount) (error, *toolchainv1alpha1.MasterUserRecord) {
+func (r *ReconcileUserAccountStatus) updateMasterUserRecord(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (*toolchainv1alpha1.MasterUserRecord, error) {
+	if userAcc.DeletionTimestamp != nil {
+		logger.Info("Updating MUR after UserAccount deletion")
+	} else {
+		logger.Info("Updating MUR")
+	}
 	cachedToolchainCluster, ok := r.getHostCluster()
 	if !ok {
-		return fmt.Errorf("there is no host cluster registered"), nil
+		return nil, fmt.Errorf("there is no host cluster registered")
 	}
 	if !cluster.IsReady(cachedToolchainCluster.ClusterStatus) {
-		return fmt.Errorf("the host cluster is not ready"), nil
+		return nil, fmt.Errorf("the host cluster is not ready")
 	}
 	mur := &toolchainv1alpha1.MasterUserRecord{}
 	name := types.NamespacedName{Namespace: cachedToolchainCluster.OperatorNamespace, Name: userAcc.Name}
-	err := cachedToolchainCluster.Client.Get(context.TODO(), name, mur)
-	if err != nil {
-		return err, nil
+	if err := cachedToolchainCluster.Client.Get(context.TODO(), name, mur); err != nil {
+		return nil, err
 	}
 	for i, account := range mur.Spec.UserAccounts {
 		if account.TargetCluster == cachedToolchainCluster.OwnerClusterName {
-			mur.Spec.UserAccounts[i].SyncIndex = userAcc.ResourceVersion
-			return cachedToolchainCluster.Client.Update(context.TODO(), mur), mur
+			if util.IsBeingDeleted(userAcc) {
+				mur.Spec.UserAccounts[i].SyncIndex = "0"
+			} else {
+				mur.Spec.UserAccounts[i].SyncIndex = userAcc.ResourceVersion
+			}
+			return mur, cachedToolchainCluster.Client.Update(context.TODO(), mur)
```

Comment:
We just need to be sure that the status will be changed in the UserAccount when it is being deleted. Otherwise, this controller won't be ever called. We have this status update there https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L147, so we should be fine.